CHANGELOG.md

```diff
@@ -81,6 +81,8 @@ Main (unreleased)
 - Flow: the `prometheus.scrape` component can now configure the scraping of
   Prometheus native histograms. (@tpaschalis)
 
+- Flow: Allow `prometheus.exporter.unix` to be specified multiple times and used in modules. (@mattdurham)
```

Comment:
This is a breaking change (though a breaking change we will all appreciate); can we document it as such?

Reply:
I'd suggest there being two changelog entries here:

* The breaking change that `prometheus.exporter.unix` now requires a label.
* The new feature (or enhancement, however it's categorized now) that `prometheus.exporter.unix` can now be specified multiple times and used in modules.

Reply:
I thought it would be a breaking change but as long as its the only `prometheus.exporter.unix` no label is needed.  Maybe this is a bug but in this case it means people can upgrade cleanly until they add another one.

This works fine if its the only one.
```
prometheus.exporter.unix {

}
```

Reply:
@mattdurham that's odd, and it _does_ seem like a bug. Labels are supposed to be required on non-singleton components, did that logic get removed?

Reply:
Seems to be specific to the exporter which leads me to believe something is hanging around that is making it behave differently. Looking into it.

Reply:
Ignore the above, you found the issue in loader where I was overzealous